### DISCUSSION TITLE
钢铁雄心4无法使用显卡

### DISCUSSION THREAD

**Discussion by notThisMoment** [Gamuxorg/bbs#414] [游戏或网站运行反馈]
游戏启动后，显卡没有调用起来，导致使用CPU渲染，然后很卡

**Comment by hyh19962008** [comment]
可以从命令行启动，然后提供一下输出信息吗
________________________________
发件人: notThisMoment ***@***.***>
发送时间: 2023年7月12日 23:55
收件人: Gamuxorg/bbs ***@***.***>
抄送: Subscribed ***@***.***>
主题: [Gamuxorg/bbs] 钢铁雄心4无法使用显卡 (Discussion #414)


游戏启动后，显卡没有调用起来，导致使用CPU渲染，然后很卡

―

**Comment by notThisMoment** [comment]
我是使用./hot4命令启动的，一开始我也想看看报错信息，但在终端没有看到有输出，是否有其他日志？

**Comment by notThisMoment** [comment]
一开始我怀疑是因为ubuntu不支持directx11，但是我在dowser里面没有看到切换使用opengl的选项

**Comment by hyh19962008** [comment]
你在设置里面看到的显示设备是llvmpipe吗？
Linux肯定不支持DirectX啊
________________________________

**Reply by notThisMoment** [reply]
在dowser里面甚至没有看到切换的设置，很奇怪。windows平台下的dowser是有的。除了dowser以外还有方法可以切换使用的渲染方式吗

**Reply by hyh19962008** [reply]
你是什么显卡啊

**Reply by hyh19962008** [reply]
1. 运行screenfetch命令，检查GPU部分是否显示llvmpipe，如果是则说明显卡驱动没有正确安装，否则应显示正确的显卡型号
2. 启动游戏后建一个新游戏，按下 ` 按钮(Esc下面)，输入rendertype查看使用的渲染器类型
3. 用intel_gpu_tools, nvidia-smi 检查一下游戏运行时的GPU占用率

**Reply by notThisMoment** [reply]
我用的是3060ti,然后nvidia-smi使用过，一点显存都没用上，

**Reply by notThisMoment** [reply]
rendertype的结果是opengl

**Reply by notThisMoment** [reply]
screenfetch命令的结果是NVIDIA Graphics Device

**Reply by notThisMoment** [reply]
会不会是opengl版本问题，我的opengl版本是OpenGL version string: 4.5 (Compatibility Profile) Mesa 20.0.8

**Comment by hyh19962008** [comment]
运行“ lsmod | nouveau ”，如果结果不是空的说明你用的是开源驱动，换成闭源驱动试试：
apt install nvidia-driver-530
________________________________

**Reply by notThisMoment** [reply]
结果是空的，没有输出

**Comment by hyh19962008** [comment]
lsmod | nvidia 呢？

Best
________________________________

**Reply by notThisMoment** [reply]
输出以下两行
nvidia 35270656 0
drm 495616 1 nvidia

**Reply by notThisMoment** [reply]
输出以下两行
nvidia    35270656     0
drm        495616          1     nvidia

**Reply by notThisMoment** [reply]
老是对不齐，注意空格哈

**Comment by hyh19962008** [comment]
感觉好像驱动不完整，screenfetch时应该能显示完整的型号的（3060），试下apt list --installed | grep nvidia-driver

Best
________________________________

**Reply by notThisMoment** [reply]
没有输出，很奇怪；只有一句警告
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

**Comment by hyh19962008** [comment]
说明你不是从软件源里安装的，你记得驱动是什么时候安装的吗？

试下执行nvidia-uninstall卸掉现在的，然后再apt install nvidia-driver-530

Best
________________________________

**Reply by notThisMoment** [reply]
驱动直接联系不上了。。。
我得分析原因先

**Comment by hyh19962008** [comment]
装完需要重启，用screenfetch和lsmod检查是否正常工作。
开不了机的话，启动时进network恢复模式，装nouveau作为亮屏驱动，然后再尝试旧一点版本的nvidia-driver


________________________________

**Reply by notThisMoment** [reply]
我用modrope命令后nvidia-smi已经可以看到显卡了，但是跑游戏的时候还是没用起来，很奇怪
